aspnetcore/tutorials/web-api-vsc.md

```diff
@@ -0,0 +1,272 @@
+
+[!code-xml[Main](web-api-vsc/sample/TodoApi/TodoApi.csproj?highlight=13-15)]
+
+Run `dotnet restore` to download and install the EF Core InMemory DB provider. You can run `dotnet restore` from the terminal or enter `⌘⇧P` or `Ctrl+Shift+P` in VS Code and then type **.NET**. Select **.NET: Restore Packages**.
+
+
+### Add a model class
+
+A model is an object that represents the data in your application. In this case, the only model is a to-do item.
+
+Add a folder named *Models*. You can put model classes anywhere in your project, but the *Models* folder is used by convention.
+
+Add a `TodoItem` class with the following code:
+
+[!code-csharp[Main](first-web-api/sample/TodoApi/Models/TodoItem.cs)]
+
+* The `[Key]` data annotation denotes the property, `Key`, is a unique identifier.
+* `[DatabaseGenerated` specifies the database will generate the key (rather than the application).
+* `DatabaseGeneratedOption.Identity` specifies the database should generate integer keys when a row is inserted. 
+
+### Create the database context
+
+The *database context* is the main class that coordinates Entity Framework functionality for a given data model. You create this class by deriving from the `Microsoft.EntityFrameworkCore.DbContext` class.
+
+Add a `TodoContext` class in the *Models* folder:
+
+[!code-csharp[Main](first-web-api/sample/TodoApi/Models/TodoContext.cs)]
+
+## Add a repository class
+
+A *repository* is an object that encapsulates the data layer. The *repository* contains logic for retrieving and mapping data to an entity model. Create the repository code in the *Models* folder.
+
+Defining a repository interface named `ITodoRepository`:
+
+[!code-csharp[Main](first-web-api/sample/TodoApi/Models/ITodoRepository.cs)]
+
+This interface defines basic CRUD operations.
+
+Add a `TodoRepository` class that implements `ITodoRepository`:
+
+[!code-csharp[Main](first-web-api/sample/TodoApi/Models/TodoRepository.cs)]
+
+Build the app to verify you don't have any compiler errors.
+
+## Register the repository
+
+By defining a repository interface, we can decouple the repository class from the MVC controller that uses it. Instead of instantiating a `TodoRepository` inside the controller we will inject an `ITodoRepository` using the built-in support in ASP.NET Core for [dependency injection](xref:fundamentals/dependency-injection).
+
+This approach makes it easier to unit test your controllers. Unit tests should inject a mock or stub version of `ITodoRepository`. That way, the test narrowly targets the controller logic and not the data access layer.
+
+In order to inject the repository into the controller, we need to register it with the DI container. Open the *Startup.cs* file. 
+
+In the `ConfigureServices` method, add the highlighted code:
+
+[!code-csharp[Main](first-web-api/sample/TodoApi/Startup.cs?name=snippet_AddSingleton&highlight=11)]
```

Comment:
Also, this requires a using for TodoAPI.Models -- for a tutorial at this level we shouldn't assume they'll figure that out.  Because of some other errors (like forgetting .cs on some file names) it took me a while to figure out what was missing.